sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuide.md

```diff
@@ -373,6 +395,123 @@ private async Task ProcessUntilCanceled(CancellationToken cancellationToken)
+
+Because the nature of legacy checkpoint data cannot be assumed due to custom storage provider support, the example uses a static set of checkpoint data for illustration.  The format of this sample data matches that used by the default Azure Storage checkpoint manager of the `EventProcessorHost` for the body of the Blob holding checkpoint data.  
+
+Reading Azure Storage data is not within the scope of this example, but details may be found in the Azure Storage documentation, see: [Quickstart: Azure Blob storage client library v12 for .NET](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-dotnet).
+
+```csharp
+public static async Task Main(string[] args)
+{
+    var blobStorageConnectionString = "<< CONNECTION STRING FOR THE STORAGE ACCOUNT >>";
+    var blobContainerName = "<< NAME OF THE BLOBS CONTAINER >>";
+    var eventHubsConnectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+    var eventHubName = "<< NAME OF THE EVENT HUB >>";
+    var consumerGroup = EventHubConsumerClient.DefaultConsumerGroupName;
+    
+    // These are stubbed for example purposes; in an actual application, these would be 
+    // the real handlers for processing events and errors in your specific scenario.
+    async Task processEventHandler(ProcessEventArgs eventArgs) => Task.CompletedTask;
+    async Task processErrorHandler(ProcessErrorEventArgs eventArgs) => Task.CompletedTask;
+    
+    // This handler will consider whether there is a legacy checkpoint for the partition available and,
+    // if so, will apply it as a default for the partition so that the processor begins at the 
+    // same location.
+    var legacyCheckpoints = ReadLegacyCheckpoints();
+    
+    Task partitionInitializingHandler(PartitionInitializingEventArgs eventArgs)
+    {
+        if (eventArgs.CancellationToken.IsCancellationRequested)
+        {
+            return Task.CompletedTask;
+        }
+    
+        if ((legacyCheckpoints.TryGetValue(eventArgs.PartitionId, out var offsetString)) 
+            && (int.TryParse(offsetString, out var offset)))
+        {
+           eventArgs.DefaultStartingPosition = EventPosition.FromOffset(offset, false);
+           Console.WriteLine($"Initialized partition: { eventArgs.PartitionId }");
+        }
+        
+        return Task.CompletedTask;
+    }
+    
+    // Create the clients and assign the handlers.
+    var storageClient = new BlobContainerClient(blobStorageConnectionString, blobContainerName);
+    var processor = new EventProcessorClient(storageClient, consumerGroup, eventHubsConnectionString, eventHubName);
+    
+    processor.PartitionInitializingAsync += partitionInitializingHandler;
+    processor.ProcessEventAsync += processEventHandler;
+    processor.ProcessErrorAsync += processErrorHandler;
+    
+    // Process until canceled.
+    try
+    {
+        while (!cancellationToken.IsCancellationRequested)
```

Comment:
super nit: I'm assuming you intentionally avoided something like below for simplicity's sake, but I think it's a tidier pattern that is well demonstrated in the tests. 
`await Task.Delay(Timeout.Infinite, cancellationSource.Token);`

Reply:
Honestly, that was a copy/paste that I didn't pay attention to.   The suggestion is a good one; applying to both areas.